include/mbedtls/config.h

```diff
@@ -2573,6 +2574,15 @@
 //#define MBEDTLS_X509_MAX_INTERMEDIATE_CA   8   /**< Maximum number of intermediate CAs in a verification chain. */
 //#define MBEDTLS_X509_MAX_FILE_PATH_LEN     512 /**< Maximum length of a path/filename string in bytes including the null terminator character ('\0'). */
 
+/**
+ * Allow SHA-1 in the default TLS configuration for certificate signing and
+ * TLS 1.2 handshake signature. Without this build-time option, SHA-1
+ * support must be activated explicitly through mbedtls_ssl_conf_cert_profile
+ * and mbedtls_ssl_conf_sig_hashes. The use of SHA-1 in TLS <= 1.1 and in
+ * HMAC-SHA-1 for XXX_SHA ciphersuites is always allowed by default.
+ */
+// #define MBEDTLS_TLS_DEFAULT_ALLOW_SHA1
```

Comment:
I think this option makes sense as we want to achieve a balance between secure defaults and compatibility (remember this is a minor release). Sadly, people are not happy when you "break" something that was "working" even if it was insecure, and even if ssl pulse sees 0% of sha-1 signed certificates, sadly there may be more still in use in the embedded world.

Also, I don't think there should be TLS_1_2 in the name as unless I'm mistaken is affects cert chain verification in all TLS versions.